### PR TITLE
Move chef from brooklyn-software-base to own module

### PR DESCRIPTION
**Pull request by aledsage**
This contains no code changes. It just moves files to the new locations, and moves snippets of some files (e.g. catalog.bom, blueprint.xml and `org.apache.brooklyn.core.resolve.entity.EntitySpecResolver`.

See https://github.com/apache/brooklyn-server/pull/826 for deletion of the code from brooklyn-software-base.

### CODE REVIEW

File: software/cm/chef/src/main/resources/catalog.bom [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+brooklyn.catalog:
+    version: "0.12.0-SNAPSHOT" # BROOKLYN_VERSION
+    itemType: entity
+    items:
+    - id: org.apache.brooklyn.entity.chef.ChefEntity
+      item:
+        type: org.apache.brooklyn.entity.chef.ChefEntity
+        name: Chef Entity
+        description: Software managed by Chef
```
Comment (tbouron) [inline software/cm/chef/src/main/resources/catalog.bom]: Shouldn't you add directly the `iconUrl` field to match the changes made in #128 ?

Comment (aledsage) [reply software/cm/chef/src/main/resources/catalog.bom]: @tbouron was planning to add iconUrl as a separate step (it needs added to `src/main/resources`, so we can reference it offline). But I'll do that now in a second commit in this PR.

**Comment by aledsage** [comment]
@tbouron changes made; ready for review + (hopefully!) merge.

@tbouron For why we have ansible, chef and salt in different modules (versus databases in one module), we don't have a hard-and-fast rule. However, with hindsight having entities like databases in the same brooklyn module was a mistake. Their release cadence and versioning are tied to that of brooklyn-core. They should have been in a community-contributed place like "brooklyncentral" in my opinion. And should have been in separate modules there.

For ansible, chef etc - though they are solving a similar problem, they are very different integrations so I think it's good to separate them into different modules.

The most compelling immediate answer though is that ansible and salt area already in different modules, so we're not going to combine them! And we should follow the same pattern for chef.

**Comment by tbouron** [comment]
@aledsage Just tested this with https://github.com/apache/brooklyn-server/pull/826, works great but requires to be rebased on the latest master + change the version in pom.xml to `0.13.0-SNAPSHOT`

After that, good to merge 👍

**Comment by aledsage** [comment]
jenkins failed with the error:
```
java.io.IOException: remote file operation failed: /home/jenkins/jenkins-slave/workspace/brooklyn-library-pull-requests at hudson.remoting.Channel@31056a0f:H25: hudson.remoting.ProxyException: hudson.maven.MavenModuleSetBuild$MavenExecutionException: org.apache.maven.project.ProjectBuildingException: Some problems were encountered while processing the POMs:
[FATAL] Non-resolvable parent POM: Could not find artifact org.apache.brooklyn:brooklyn-library:pom:0.13.0-SNAPSHOT and 'parent.relativePath' points at wrong local POM @ line 30, column 13
```
I presume this is temporary, until jenkins has built brooklyn-library master etc, so that 0.13.0-SNAPSHOT of everything is in apache's maven snapshot repository.

Will kick off a "retest this please" again later.

**Comment by tbouron** [comment]
> jenkins failed with the error:

See my previous comment

**Comment by aledsage** [comment]
retest this please

**Comment by tbouron** [comment]
retest this please

**Comment by aledsage** [comment]
retest this please

**Pull request merged**